libslide/src/parser.rs

```diff
@@ -89,35 +91,34 @@ where
         exp_term:            lhs = num_term,            rhs = exp_term,            op = [Exp]
     );
 
-    fn num_term(&mut self) -> Box<Self::Expr> {
+    fn num_term(&mut self) -> Rc<Self::Expr> {
         if let Some(Ok(op)) = self.input().peek().map(UnaryOperator::try_from) {
             self.input().next();
-            return Box::new(
-                UnaryExpr {
-                    op,
-                    rhs: self.exp_term(),
-                }
-                .into(),
-            );
+            let node = UnaryExpr {
+                op,
+                rhs: self.exp_term(),
+            }
+            .into();
+            return self.finish_expr(node);
         }
```

Comment:
is this doable without creating a temp var

Reply:
Yes but it's a bunch of nested calls that spans >5 lines, which I would prefer to avoid. This will definitely get optimized out anyway